src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -789,5 +789,55 @@ internal enum GCConfigurationType
 
         [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "GCInterface_EnumerateConfigurationValues")]
         internal static unsafe partial void _EnumerateConfigurationValues(void* configurationDictionary, delegate* unmanaged<void*, void*, void*, GCConfigurationType, long, void> callback);
+
+        private static int _RefreshMemoryLimit()
+        {
+            ulong specifiedFlags = 0;
+            ulong? heapHardLimit = AppContext.GetData("GCHeapHardLimit") as ulong?; if (heapHardLimit != null) { specifiedFlags |= 1; } else { heapHardLimit = 0; }
+            ulong? heapHardLimitPercent = AppContext.GetData("GCHeapHardLimitPercent") as ulong?; if (heapHardLimitPercent != null) { specifiedFlags |= 2; } else { heapHardLimitPercent = 0; }
+            ulong? heapHardLimitSOH = AppContext.GetData("GCHeapHardLimitSOH") as ulong?; if (heapHardLimitSOH != null) { specifiedFlags |= 4; } else { heapHardLimitSOH = 0; }
+            ulong? heapHardLimitLOH = AppContext.GetData("GCHeapHardLimitLOH") as ulong?; if (heapHardLimitLOH != null) { specifiedFlags |= 8; } else { heapHardLimitLOH = 0; }
+            ulong? heapHardLimitPOH = AppContext.GetData("GCHeapHardLimitPOH") as ulong?; if (heapHardLimitPOH != null) { specifiedFlags |= 16; } else { heapHardLimitPOH = 0; }
+            ulong? heapHardLimitSOHPercent = AppContext.GetData("GCHeapHardLimitSOHPercent") as ulong?; if (heapHardLimitSOHPercent != null) { specifiedFlags |= 32; } else { heapHardLimitSOHPercent = 0; }
+            ulong? heapHardLimitLOHPercent = AppContext.GetData("GCHeapHardLimitLOHPercent") as ulong?; if (heapHardLimitLOHPercent != null) { specifiedFlags |= 64; } else { heapHardLimitLOHPercent = 0; }
+            ulong? heapHardLimitPOHPercent = AppContext.GetData("GCHeapHardLimitPOHPercent") as ulong?; if (heapHardLimitPOHPercent != null) { specifiedFlags |= 128; } else { heapHardLimitPOHPercent = 0; }
+            if (specifiedFlags != 0)
+            {
+                GCHeapHardLimitInfo heapHardLimitInfo = new GCHeapHardLimitInfo
+                {
+                    HeapHardLimit = heapHardLimit.Value!,
+                    HeapHardLimitPercent = heapHardLimitPercent.Value!,
+                    HeapHardLimitSOH = heapHardLimitSOH.Value!,
+                    HeapHardLimitLOH = heapHardLimitLOH.Value!,
+                    HeapHardLimitPOH = heapHardLimitPOH.Value!,
+                    HeapHardLimitSOHPercent = heapHardLimitSOHPercent.Value!,
+                    HeapHardLimitLOHPercent = heapHardLimitLOHPercent.Value!,
+                    HeapHardLimitPOHPercent = heapHardLimitPOHPercent.Value!,
+                    SpecifiedFlags = specifiedFlags
```

Comment:
>SpecifiedFlags [](http://example.com/codeflow?start=20&length=14)

we could just have an init value of -1 for these so you know if they are set by the appcontext. and if it's not -1,it means it's set